azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/utils/ResourceManagerThrottlingInterceptor.java

```diff
@@ -26,15 +29,17 @@
  */
 public class ResourceManagerThrottlingInterceptor implements Interceptor {
     private static final String LOGGING_HEADER = "x-ms-logging-context";
-    private static final ReentrantLock REENTRANT_LOCK = new ReentrantLock();
+    private static final ConcurrentMap<String, ReentrantLock> REENTRANT_LOCK_MAP = new ConcurrentHashMap<>();
 
     @Override
     public Response intercept(Chain chain) throws IOException {
         // Gate control
+        String subscriptionId = ResourceUtils.subscriptionFromResourceId(chain.request().url().url().getPath());
+        REENTRANT_LOCK_MAP.putIfAbsent(subscriptionId, new ReentrantLock());
         try {
-            synchronized (REENTRANT_LOCK) {
-                if (REENTRANT_LOCK.isLocked()) {
-                    REENTRANT_LOCK.wait();
+            synchronized (REENTRANT_LOCK_MAP.get(subscriptionId)) {
```

Comment:
It looks like this is synchronizing on both the intrinsic lock and the ReentrantLock methods, which seems off. But if it works, it works.